sdk/nodejs/cmd/run/run.ts

```diff
@@ -53,9 +64,38 @@ function projectRootFromProgramPath(program: string): string {
     }
 }
 
+/**
+  * @internal
+  * This function searches for the nearest package.json file, scanning up from the
+  * program path until it finds one. If it does not find a package.json file, it
+  * it returns the folder enclosing the program.
+  * @param programPath the path to the Pulumi program; usually the current working directory.
```

Comment:
Not "the current working directory", this is a mistake we keep making 😆 
It should be the project "main" directory, which defaults to project "root" directory.

Reply:
Good point!